flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2416,6 +2420,44 @@ static mlir::Value adjustedArrayElement(mlir::Location loc,
   return builder.createConvert(loc, adjustedArrayElementType(eleTy), val);
 }
 
+/// Helper to generate calls to scalar user defined assignment procedures.
+static void genScalarUserDefinedAssignmentCall(fir::FirOpBuilder &builder,
+                                               mlir::Location loc,
+                                               mlir::FuncOp func,
+                                               mlir::Value lhs,
+                                               mlir::Value rhs) {
+  auto prepareUserDefinedArg = [](fir::FirOpBuilder &builder,
+                                  mlir::Location loc, mlir::Value value,
+                                  mlir::Type argType) -> mlir::Value {
+    // In general, we may not have the rhs length threaded here, since only
+    // the destination length parameters are being tracked.
```

Comment:
It looks like this is being called from a continuation in the evaluate loop. The IterationSpace ought to have a proper ExtendedValue for what the rhs was evaluated to be (`elementExv()`), which would include the LEN (modulo bugs). Also, it looks like the `lhs` argument is the `array_merge` operation, which should have been created with the appropriate type parameters properties. Is something else required?